helpers/lib/common.js

```diff
@@ -0,0 +1,38 @@
+function isObject(val) {
+    return val !== null && typeof val === 'object';
+}
+
+function isOptions(val) {
+    return isObject(val) && val.fn;
+}
+
+function isString(val) {
+    return typeof val === 'string';
+}
+
+function isEmptyObject(val) {
+    return isObject(val) && Object.keys(val).length === 0;
+}
+
+function isTruthy(val) {
+    if (Array.isArray(val)) {
+        return !!val.length;
+    } else if (isEmptyObject(val)) {
+        return false;
+    } else {
+        return !!val;
+    }
+}
+//function to check if two values are equal also does a check for NaN === NaN
+function equals(val1, val2) {
+    return val1 === val2 || (val1 !== val1 && val2 !== val2);
```

Comment:
Can we use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is for this?

Reply:
Learn something new everyday.  I believe we can.

Reply:
fixed.